frontend/src/settings/Preferences.jsx

```diff
@@ -1,16 +1,89 @@
 import * as React from 'react';
 import {
-  Text,
+  Alert, Button, Switch, Platform,
 } from 'react-native';
+import { LinearGradient } from 'expo-linear-gradient';
+import RNDateTimePicker from '@react-native-community/datetimepicker';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useFocusEffect } from '@react-navigation/native';
 import {
-  BoilerplateView,
+  PreferencesView, CardTitle, PreferencesTitle,
 } from '../Themes';
 
 function Preferences() {
+  const [darkMode, setDarkMode] = React.useState(false);
+  const [Alarm, setAlarm] = React.useState(true);
+  const [twelveHour, setTwelveHour] = React.useState(true);
+  const [alarmTime, setAlarmTime] = React.useState(new Date());
+  const [show, setShow] = React.useState(false);
+
+  const onChange = (event, selectedDate) => {
+    const currentDate = selectedDate || alarmTime;
+    setShow(Platform.OS === 'ios');
```

Comment:
Setting whether or not the alarm shows based on OS should not be done here. Because let's say you click on the `Set Alarm` button and the picker shows up and that's all great. But when you click on it and change the value, if you're not on iOS, then the picker will just disappear without any explanation at all. Checking for something like this should be done earlier in the logic. Two places I can see where this would be better implemented is when the page first loads, if the OS can't support alarms or the picker or whatever, then don't display the alarm button or have anything related to the alarm appear on screen. The other place this can be implemented is in the `Set Alarm` button, when you click it maybe it checks there and if your OS is not supported it can popup a warning and then not let you continue.

Reply:
so Android has its own ok/cancel button to close the date-time picker. And without that show based on the change, the picker will persist even after the user has pressed ok/cancel. So the OS can support the alarm/picker, it's just that the picker should be 'hidden' on change because the user will then navigate out themselves.

Reply:
Interesting. That's fine then. Thanks for clearing that up.